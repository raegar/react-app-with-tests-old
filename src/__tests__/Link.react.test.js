import React from 'react';
import Link from '../Link.react';
import renderer from 'react-test-renderer';

test('Link changes the class when hovered', () => {
    const component = renderer.create(
        <Link page="http://www.ucp.ac.uk">University Centre Peterborough</Link>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseEnter();
    //re-rendering now class should be STATUS.HOVERED
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    
    // manually trigger the callback
    tree.props.onMouseLeave();
    // re-rendering now class hsould be STATUS.NORMAL
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();


})