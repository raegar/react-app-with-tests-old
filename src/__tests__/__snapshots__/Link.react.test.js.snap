// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="http://www.ucp.ac.uk"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  University Centre Peterborough
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="http://www.ucp.ac.uk"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  University Centre Peterborough
</a>
`;

exports[`Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="http://www.ucp.ac.uk"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  University Centre Peterborough
</a>
`;
